### DISCUSSION TITLE
全象 2022 Q1 路线图

### DISCUSSION THREAD

**Discussion by hqc19907228** [quanxiang-cloud/quanxiang#23] [Announcements]
大家好，不知不觉 2021 已经悄然离去，我们又迎来了崭新的一年。新年伊始，QuanXiang 开源社区的成员们将向大家宣布 QuanXiang 在 2022 Q1 的目标和计划！

# QuanXiang
首先，请允许我向大家简单的介绍一下 QuanXiang。

QuanXiang 是一个基于云原生的、完全容器化的开源低代码平台，是全象云低代码平台的开源版本，用于辅助构建企业各类数字化应用。平台目前提供云上无代码和低代码两种应用开发模式，支持可视化设计，让开发人员和业务用户能够通过简单的拖拽、参数配置等方式快速完成应用开发。作为一个集低代码开发能力、身份认证能力、容器 DevOps 能力于一体的多应用集成和管理平台，QuanXiang 支持快速构建应用、便捷维护管理应用、企业存量业务及全象云构建业务的集成。

# 平台功能
QuanXiang 围绕应用设计、开发、部署、运维全生命周期管理，为企业的信息化数智化建设赋能。平台核心能力如下：

### 快速应用开发
- 可视化设计器：用户通过简单的拖拽、参数配置等方式就能完成页面设计、工作流编排、数据模型设计和角色权限的定义。
- 表单引擎：系统提供丰富的页面组件，能够满足页面呈现的自定义组件需求。
- 工作流引擎：包含灵活的触发方式和丰富的流程组件。提供审批、填写等人为节点处理，支持数据新增、数据更新等自动流程节点处理。同时提供规则引擎的能力，满足复杂业务下的逻辑定义。

### 多云部署和运维
- QuanXiang 基于 Kubernetes 的容器化部署方案，CI/CD 持续交付部署，为应用的平滑部署、稳定运行保驾护航，大大降低了应用上线后迭代升级的风险和复杂度。
- 支持不同云厂商的部署及运维。
- 平台提供系统日志，支持查看所有操作记录。

### 多端兼容适配
- 平台应用均可一次设计，在多端灵活适配，目前支持一键发布为 Web 应用。
- 表单引擎支持一键发布为 Native APP。

### 灵活组织管理
- 企业通讯录：提供多种管理通讯录方式，帮助企业快速完成组织的构建。
- 角色管理：企业角色权限按需细分，保障平台账户访问安全和数据安全。

### 系统连接能力
- 平台内部支持应用内的数据连接，提供了不同粒度的数据连接能力，包括：表与表之间的数据联动更新、字段与字段间的联动交互。

### 可插拔的架构
- QuanXiang 是云原生、分布式架构的平台系统。各个核心服务（聚合类服务除外）采用完全解耦的、低内聚的方式设计，服务之间通过 API 接口进行访问。

# 路线图
在 2022 Q1 ，QuanXiang 将会专注于平台应用开发能力的完善。平台将会新增自定义页面功能，支持应用独立访问，支持构建 Web 端以及简易版本的 Native APP。同时将大幅优化平台的可用性、易用性、稳定性和可维护性。具体将完善的功能包括：

### 工作流能力完善
- 添加定时任务触发功能，解决日常使用中定时任务管理的需求。
- 完善流程变量在工作流中的应用，解决流程引擎中数据的流转问题。
- 添加 Webhook 节点，满足推送、自定义 API 的请求。
- 完善更新节点数据的能力，支持用户对触发流程的表单数据及对应的子表单数据进行修改。

### 新增自定义页面的能力（页面引擎）
- 新增页面引擎支持的数据源类型，对组件数据来源进行统一管理。
    - 自定义变量：通过定义中间变量的方式使组件与变量形成绑定关系，当变量发生变化后，组件即重新进行渲染。
    - API 请求：通过定义 API 变量（此处 API 变量为变量中的一种），使 API 与组件形成绑定关系，API 发送请求，当接收的数据发生变化后，组件即重新进行渲染。
- 新增页面引擎的组件：包括文本、图片、按钮、容器、输入框、链接文字、iframe。
- 支持修改页面及组件的样式：包括但不限于位置、大小、背景颜色、字体等样式。
- 支持组件数据的循环渲染：提供当 API 返回多条数据后，渲染成为多条数据的能力。
- 支持以大纲树的形式展示整个页面的组件层级，可通过选中大纲树中的组件对页面中的组件进行定位。
- 支持页面生命周期管理：可在页面加载与关闭时定义 JavaScript 语法逻辑。

### 其它
- 单应用管理：支持应用独立访问，满足用户直接访问应用的需求，不受平台其它应用的干扰。包括应用登录、审批、填写等，均可按照单应用进行功能管理。
- 应用 URL 设置：除支持平台 URL 访问外，还将支持应用独立 URL 访问。

以上就是 QuanXiang 2022 Q1 的开发计划，如果你有不同的想法，可以在公众号内给我们发送消息，我们期待所有来自社区的所有意见和建议！

**Comment by jyRMc** [comment]
我司有想接入贵公司零代码平台，公众号发送了消息没人回复，方便提供联系方式沟通一下吗？

**Reply by hqc19907228** [reply]
> 我司有想接入贵公司零代码平台，公众号发送了消息没人回复，方便提供联系方式沟通一下吗？

您好，您可以把您这边的想法发个邮件给我，我让我们的同事先看一下！我的邮箱：qichonghan@yunify.com

**Comment by hqc19907228** [comment]
大家好，我的邮箱：qichonghan@yunify.com 已经不能使用了，大家如果有相关问题想要咨询的话，为了避免遗漏，还是尽量在讨论区进行留言哈！
